catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -167,42 +108,44 @@ public UpdateResponse process(UpdateResponse input) throws PluginExecutionExcept
      * Removes deleted metacards from the file system backup.
      *
      * @param input the {@link DeleteResponse} to process
-     * @return
-     * @throws PluginExecutionException
+     * @return {@link DeleteResponse}
      */
     @Override
-    public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
-        if (enableBackupPlugin && Requests.isLocal(input.getRequest())) {
-            LOGGER.debug("Deleting metacards contained in DeleteResponse from backup.");
+    public DeleteResponse process(DeleteResponse input) {
 
-            if (rootBackupDir == null) {
-                throw new PluginExecutionException("No root backup directory configured.");
-            }
-
-            List<String> errors = new ArrayList<String>();
+        execute(() -> delete(input.getDeletedMetacards()));
+        return input;
+    }
 
-            List<Metacard> metacards = input.getDeletedMetacards();
+    /**
+     * Sets the number of subdirectory levels to create. Two characters from
+     * each metacard ID will be used to name each subdirectory level.
+     *
+     * @param levels number of subdirectory levels to create
+     */
+    public void setSubDirLevels(int levels) {
+        this.subDirLevels = levels;
```

Comment:
We should do input validation on `levels` and throw an `IllegalArgumentException` if the value is < 0.